### ISSUE TITLE
2.1.0-beta.2 renders blank UI

### ISSUE THREAD

**Issue by henrikjohansen** [Graylog2/graylog2-server#2653]
After upgrading one of our test machines to 2.1.0-beta.2 graylog starts just fine and presents it's login page. After login however the UI does not get rendered.

Chrome's Javascript Console shows a few JS errors:

<img width="648" alt="screen shot 2016-08-10 at 15 25 55" src="https://cloud.githubusercontent.com/assets/1473308/17555824/4e9f9ccc-5f11-11e6-9678-1a2d06af7ddf.png">

## Your Environment

```
2016-08-10T15:21:02.196+02:00 INFO  [ServerBootstrap] Graylog server 2.1.0-beta.2+ffa3355 starting up
2016-08-10T15:21:02.196+02:00 INFO  [ServerBootstrap] JRE: Oracle Corporation 1.8.0_101 on Linux 2.6.32-573.22.1.el6.x86_64
2016-08-10T15:21:02.197+02:00 INFO  [ServerBootstrap] Deployment: rpm
2016-08-10T15:21:02.197+02:00 INFO  [ServerBootstrap] OS: CentOS release 6.8 (Final)
2016-08-10T15:21:02.197+02:00 INFO  [ServerBootstrap] Arch: amd64
```
server.conf : 
```
is_master = true
node_id_file = /etc/graylog/server/node-id
password_secret = ...
root_password_sha2 = ...
root_timezone = Europe/Copenhagen
plugin_dir = /usr/share/graylog-server/plugin
rest_listen_uri = http://1.2.3.4:12900/
web_listen_uri = http://1.2.3.4:9000
rotation_strategy = count
elasticsearch_max_docs_per_index = 20000000
rotation_strategy = count
elasticsearch_max_docs_per_index = 20000000
elasticsearch_max_number_of_indices = 20
retention_strategy = delete
elasticsearch_max_number_of_indices = 20
retention_strategy = delete
elasticsearch_shards = 4
elasticsearch_replicas = 0
elasticsearch_index_prefix = graylog
allow_leading_wildcard_searches = false
allow_highlighting = false
elasticsearch_analyzer = standard
output_batch_size = 500
output_flush_interval = 1
output_fault_count_threshold = 5
output_fault_penalty_seconds = 30
processbuffer_processors = 5
outputbuffer_processors = 3
processor_wait_strategy = blocking
ring_size = 65536
inputbuffer_ring_size = 65536
inputbuffer_processors = 2
inputbuffer_wait_strategy = blocking
message_journal_enabled = true
message_journal_dir = /var/lib/graylog-server/journal
lb_recognition_period_seconds = 3
mongodb_uri = mongodb://localhost/graylog
mongodb_max_connections = 1000
mongodb_threads_allowed_to_block_multiplier = 5
content_packs_dir = /usr/share/graylog-server/contentpacks
content_packs_auto_load = grok-patterns.json
```

**Comment by edmundoa** [comment]
Hi @henrikjohansen,

I guess you already tried, but just to be sure: do you see the problem in other browsers or using an incognito window?

Also, does the error happen all the time or only after log-in?

Thank you!

**Comment by hc4** [comment]
I had same problem until pressed Ctrl+F5 to clear caches.

**Comment by henrikjohansen** [comment]
Hi @edmundoa,

The error appears on both Chrome (with and without incognito mode) and Safari. If I manually try to navigate to other pages /streams or /system/overview the page stays blank.

**Comment by edmundoa** [comment]
I have tried to reproduce the issue with the beta 2 tarball in OSX but I couldn't, I'll take a look in CentOS. In the meantime, you mentioned you upgrade the test machine to 2.1.0-beta.2, from which version did you upgrade?

**Comment by henrikjohansen** [comment]
Upgrade was from 2.0.3 - the graylog packages were removed and afterwards the journal files and mongo databases were deleted followed by a reinstall of the new graylog packages.

**Comment by edmundoa** [comment]
I just managed to finish a small setup in a CentOS 6 VM and I couldn't reproduce the issue. I basically did something similar to the upgrade you described: first installed 2.0.3, remove the packages, and install 2.1.0-beta.2.

Would you be so kind as to share with us the plugin versions you see while starting Graylog server? Additionally, are you using some kind of proxy on that setup that may be caching the JS files? My best guess right now is that your browser is still getting an old JS file, either from an outdated plugin or from some cache.

**Comment by henrikjohansen** [comment]
```
2016-08-10T15:20:45.251+02:00 INFO  [ServiceManagerListener] Services are now stopped.
2016-08-10T15:20:50.491+02:00 INFO  [CmdLineTool] Loaded plugin: MetricsOutput 1.1.4 [org.graylog.outputs.metrics.MetricsOutputPlugin]
2016-08-10T15:20:50.496+02:00 INFO  [CmdLineTool] Loaded plugin: Elastic Beats Input 1.1.0-beta.2 [org.graylog.plugins.beats.BeatsInputPlugin]
2016-08-10T15:20:50.498+02:00 INFO  [CmdLineTool] Loaded plugin: Collector 1.1.0-beta.2 [org.graylog.plugins.collector.CollectorPlugin]
2016-08-10T15:20:50.499+02:00 INFO  [CmdLineTool] Loaded plugin: Enterprise Integration Plugin 1.1.0-beta.2 [org.graylog.plugins.enterprise_integration.EnterpriseIntegrationPlugin]
2016-08-10T15:20:50.500+02:00 INFO  [CmdLineTool] Loaded plugin: MapWidgetPlugin 1.1.0-beta.2 [org.graylog.plugins.map.MapWidgetPlugin]
2016-08-10T15:20:50.500+02:00 INFO  [CmdLineTool] Loaded plugin: Pipeline Processor Plugin 1.0.0-beta.2 [org.graylog.plugins.pipelineprocessor.ProcessorPlugin]
2016-08-10T15:20:50.501+02:00 INFO  [CmdLineTool] Loaded plugin: Anonymous Usage Statistics 2.1.0-beta.2 [org.graylog.plugins.usagestatistics.UsageStatsPlugin]
2016-08-10T15:20:50.646+02:00 INFO  [CmdLineTool] Running with JVM arguments: -Xms1g -Xmx1g -XX:NewRatio=1 -XX:+ResizeTLAB -XX:+UseConcMarkSweepGC -XX:+CMSConcurrentMTEnabled -XX:+CMSClassUnloadingEnabled -XX:+UseParNewGC -XX:-OmitStackTraceInFastThrow -Djava.library.path=/usr/share/graylog-server/lib/sigar -Dlog4j.configurationFile=file:///etc/graylog/server/log4j2.xml -Dgraylog2.installation_source=rpm
2016-08-10T15:20:54.134+02:00 INFO  [InputBufferImpl] Message journal is enabled.
```

There is no proxy in front of that machine ... or between my machine and that test server.

Just for the heck of it I tried clearing out browser cache etc ... the UI still won't render.

**Comment by edmundoa** [comment]
The pipeline-processor plugin is outdated in your setup, it should be 1.1.0-beta.2.
```
2016-08-10T15:20:50.500+02:00 INFO  [CmdLineTool] Loaded plugin: Pipeline Processor Plugin 1.0.0-beta.2 [org.graylog.plugins.pipelineprocessor.ProcessorPlugin]
```

I'm not sure the cause of this, but I think if you update it, it should be fine again. The easiest way is to grab it from the beta.2 tarball, it should be in the `plugin` folder.

**Comment by henrikjohansen** [comment]
Indeed - there was a rogue ```graylog-plugin-pipeline-processor-1.0.0-beta.3-SNAPSHOT.jar```  file (which I got from Kay while debugging a problem with the grok pipeline function) that was not removed during the uninstall since it was not part of the original package.

Sorry for the confusion :(

**Comment by edmundoa** [comment]
No worries, glad to see it is gone now! 👍

**Comment by hc4** [comment]
But still it seems bad, that graylog not working at all because of problems in plugin

**Comment by henrikjohansen** [comment]
Ideally Graylog would only load internal plugins that match the release version :)

**Comment by edmundoa** [comment]
I suggest to open a new ticket for that, in that way we can keep it in mind :)
